### PR TITLE
Documentation library évolution to 3.5

### PR DESCRIPTION
**Pull request by mbaillif**

### CODE REVIEW

**Comment by MarcinOrlowski** [comment]
Could you please put summary description what this PR is changing? I know I can see thru all 512 (!) files, but I would like to know your intent. For a moment thought you renamed the app to contain `évolution` (which would be wrong). Also your commit messages do note tell much not to mention you touched 512 files. That PR should really be split into smaller units unless changes in all 512 files are exactly of the same nature (which is not the case)

**Comment by mbaillif** [comment]
Indeed, I am sorry for my lack of communication.
In fact, I got it into my head to systematically adapt all the documentation from the bookstores that I had previously left behind.
- Verification of compliance with version 3.5
- Standardization of HTML code with CSS to make it compatible with javahelp2.0 and current browsers
- Improvement of the figures and the aesthetics of the pages.
- This is done with the least possible modification of what is already written. I sometimes add missing points.

Currently I'm working on the standardization of the HTML code and the aesthetics of the pages.
The documentation pages of the library are very repetitive and in many cases, in order to be efficient, I use functions like search/replace on all the files.
For example: I had to change <div class=maindiv> to <div class="maindiv"> or <q>Orientation</q> by <b class="propertie">Orientation</b> this leads to change many files maybe only for one line.
In other cases I completely reorganize the HTML page (indentation, line feed...) this too more or less automatically. In this case it completely modifies the file (according to github) but in reality not really.
Afterwards, as I am not a robot, sometimes I let myself translate a page or two into French, just to recharge my batteries :-).

I will try to do a little better.
But there will still be some PR like these, they should decrease because I'm still making some progress in these tasks.
I should be done by the end of the year.

I do this work in my travels and I confess that once in the texts I do not remain as systematic as I would like.

I wish you a great day

**Comment by MarcinOrlowski** [comment]
Please do not add another commits for a while a wait while we review this one. It's really huge now.

File: src/main/resources/doc/en/html/libs/arith/index.html [hunk 1]
```diff
@@ -101,11 +101,75 @@ <h1>
...
+            <a href="divider.html"><img  class="iconlibs" src="../../../../icons/6464/divider.png" alt="#########" width="32" height="32" border="0"></a>
+          </td><td>
+            &nbsp;
+          </td><td>
+            <a href="fpdivider.html">Floating Point Divider</a>
+          </td>
+        </tr>
+        <tr>
+          <td align="right">
+            <a href="negator.html"><img  class="iconlibs" src="../../../../icons/6464/negator.png" alt="#########" width="32" height="32" border="0"></a>
```
Comment (MarcinOrlowski) [inline src/main/resources/doc/en/html/libs/arith/index.html:143]: Please remove `alt="#########"` (multiple places). This is absolute no-no. You do not have any content for `alt` then omit it. But do not add content that serves no purpose.

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]
I did quick look at the changes but I was not able to do **proper** review. In fact, if I'd be in charge then this PR would be rejected simply because it changes multiple things in completely different "domains". For example, if you quote certain HTML values, then this should be one PR. If you add some content about the components, this should be another. If you change CSS, if should be yet another PR. 

It'd be good if others (@BFH-ktt1, @maehne , @davidhutchens ) could take a look and share their thoughts, esp. knowing I did not check any of the added texts (i.e. about certain components) etc. I just looked at the diff roughly. Let me be frank - I do not mind PRs that touches 500 files as I submitted massive PRs myself occasionally in the past, but such thing only makes sense **[IFF](https://www.abbreviations.com/term/3333)** the changes are of the same type. If not, it just creates massive reviewers' hell.

To some how sum this up:thanks for the efforts @mbaillif, the more hands on desk the better, but in this particular case, could you please see if you could split this PR into smaller ones (best, by "domain" / type of changes)? It would be a waste to drop your changes as some (even all - as I said, I haven't checked all deeply) make sense, but this PR is not a nicest one and it is hard to tell if merging it will improve things or improve here but wreck elsewhere. Hope you do understand my concerns.

**Comment by MarcinOrlowski** [comment]
Please do not add another commits for a while a wait while we review this one. It's really huge now.

File: src/main/resources/doc/de/html/guide/about/indexa.html [hunk 2]
```diff
@@ -12,7 +12,7 @@
     <link rel="stylesheet" type="text/css" href="../../style.css">
   </head>
   <body>
-    <div class=maindiv>
+    <div class="maindiv">
```
Comment (R3dst0ne) [inline src/main/resources/doc/de/html/guide/about/indexa.html:15]: Could you please do a separate PR where only this change is done?
When we merge it, this PR will become much cleaner and easier to review.

File: src/main/resources/doc/en/html/libs/arith/index.html (hunk 1 above)
Comment (mbaillif) [reply src/main/resources/doc/en/html/libs/arith/index.html:143]: For the #####, it is a verification tool that adds them, I thought to put a good information, this in a later stage.
It bothers me too, but it is a detail.

**Comment by mbaillif** [comment]
If it was code obviously this PR is not good and I understand your concern.
However, it is only about the documentation and mainly about its formatting.
You can observe the gain by opening, for example, version 2.13 and go to the help / userguide
then open the 3.5 and do the same.

You can also observe the modifications of the library doc on this site www.baillifard.com\logisim on your phone for example.
There are also two documents at the root of the doc, which give the state of this documentation.
You can access all pages simply by clicking in the boxes.
In red what is not there.
In light green where there is work.
In dark green or it is acceptable.
If there is a V, it needs a proofreading correction.
If there is a C, something needs to be added.

And indeed there is work for a revision. which I would appreciate.

It is not finished, and there is still a lot to do and it is not perfect. 
There are still many technical and textual imperfections, but less and less.

I haven't seen many people edit these documents in the last three years. 

On my fork: I have almost finished (my participation) on the wiring part. I'm thinking of starting the Gate part.
(Note: there is an error in the transistor truth table, it is corrected).

I will make a proposal in a second message later. 
Please take the time to do what I propose above.

I will not be able to connect in the next days.
I hope to provide this proposal before.

File: src/main/resources/doc/de/html/guide/about/indexa.html (hunk 2 above)
Comment (mbaillif) [reply src/main/resources/doc/de/html/guide/about/indexa.html:15]: thank you for your advice
